File: sdks/ruby/test/remote_smoke.rb
# Remote-GraphQL e2e smoke for the Ruby SDK's pure-stdlib client
# (Cvisor::RemoteSandbox over net/http — no libcvisor). Point it at a running
# cvisord:
#   CVISOR_GRAPHQL_URL=http://127.0.0.1:8080/graphql CVISOR_TOKEN=... ruby test/remote_smoke.rb
$LOAD_PATH.unshift File.expand_path("../lib", __dir__)
require "cvisor/remote"

def assert(cond, msg)
  raise "assertion failed: #{msg}" unless cond
end

url = ENV.fetch("CVISOR_GRAPHQL_URL", "http://127.0.0.1:8080/graphql")
token = ENV.fetch("CVISOR_TOKEN", "")

remote = Cvisor::RemoteSandbox.new(url, token)

health = remote.health
assert(health["ok"] == true, "health not ok: #{health.inspect}")

out = remote.run("echo hello")
assert(out["stdout"] == "hello\n", "run stdout: #{out.inspect}")
assert(out["exitCode"] == 0, "run exit code: #{out.inspect}")

sb = remote.create_sandbox
assert(!sb["id"].to_s.empty?, "create_sandbox returned no id: #{sb.inspect}")
remote.write_file(sb["id"], "/tmp/data.txt", "round-trip\n")
data = remote.read_file(sb["id"], "/tmp/data.txt")
assert(data == "round-trip\n", "read_file round-trip: #{data.inspect}")
remote.free_sandbox(sb["id"])

puts "RUBY_GRAPHQL_OK"

File: sdks/rust/examples/remote_smoke.rs
//! Remote-GraphQL e2e smoke for the Rust SDK's pure-HTTP client
//! (`cvisor::RemoteSandbox` over ureq — no libcvisor). Run against a running
//! cvisord:
//!
//! ```sh
//! CVISOR_GRAPHQL_URL=http://127.0.0.1:8080/graphql CVISOR_TOKEN=... \
//!   cargo run --example remote_smoke
//! ```

use std::env;

use cvisor::RemoteSandbox;

fn main() {
    let url = env::var("CVISOR_GRAPHQL_URL")
        .unwrap_or_else(|_| "http://127.0.0.1:8080/graphql".to_string());
    let token = env::var("CVISOR_TOKEN").unwrap_or_default();

    let mut remote = RemoteSandbox::new(url, token);

    let health = remote.health().expect("health");
    assert!(health.ok, "health not ok: {health:?}");

    let out = remote.run("echo hello", Default::default()).expect("run");
    assert_eq!(out.stdout, "hello\n", "run stdout: {out:?}");
    assert_eq!(out.exit_code, 0, "run exit code: {out:?}");

    let info = remote.create_sandbox("").expect("create_sandbox");
    assert!(!info.id.is_empty(), "create_sandbox returned no id");
    remote
        .write_file("/tmp/data.txt", b"round-trip\n")
        .expect("write_file");
    let data = remote.read_file("/tmp/data.txt").expect("read_file");
    assert_eq!(data, b"round-trip\n", "read_file round-trip");
    remote.free_sandbox().expect("free_sandbox");

    println!("RUST_GRAPHQL_OK");
}
